Docs: Set up alert state history (#71123)
grafana/grafana: docs/sources/alerting/set-up/configure-alert-state-history/index.md

This applies to Open Source only. There is no configuration required if you are using Grafana Cloud.
{{% /admonition %}}

Starting with Grafana 10, Alerting can also record all of the alert rule state changes for your Grafana managed alert rules to a Loki instance.

This allows you to explore the behavior of your alert rules in the Grafana explore view and levels up the existing state history modal with a powerful new visualisation.

<!-- image here, maybe the one from the blog? -->

## Configuring Loki

To set up alert state history we have to make sure we have a Loki instance we can write data to. The default settings might need some tweaking, the state history modal might query up to 30 days of data.

The following change to the default configuration should work for most instances, but we recommend looking at the full Loki configuration settings and adjust according to your needs.

As this might have a performance impact on an existing Loki instance we also recommend a separate Loki instance for Alert state history if this is of concern to you.

```yaml
limits_config:
  split_queries_by_interval: '24h'
  max_query_parallelism: 32
```

## Configuring Grafana

> Next up we need to configure Alert state history in Grafana. To do so we'll need to add some additional configuration to its configuration file.

Reviewer: ```suggestion
We need some additional configuration in the Grafana configuration file to have it working with the alert state history.
```

> The following example below will instruct Grafana to write alert state history to a local Loki instance.

Reviewer: ```suggestion
The example below instructs Grafana to write alert state history to a local Loki instance:
```

```toml
[unified_alerting.state_history]
enabled = true
backend = "loki"
loki_remote_url = "http://localhost:3100"

[feature_toggles]
enable = alertStateHistoryLokiSecondary, alertStateHistoryLokiPrimary, alertStateHistoryLokiOnly
```

<!-- TODO can we add some more info here about the feature flags and the various different supported setups with Loki as Primary / Secondary, etc? -->

## Adding the Loki data source

See our instructions on [adding a data source](/docs/grafana/latest/administration/data-source-management/).

## Querying the history

If everything is set up correctly you can use the Grafana Explore view to start querying the Loki data source.

> A simple litmus test to see if data is being written correctly into the Loki instance is by running the following query:

Reviewer: ```suggestion
A simple litmus test to see if data is being written correctly into the Loki instance is the following query:
```
